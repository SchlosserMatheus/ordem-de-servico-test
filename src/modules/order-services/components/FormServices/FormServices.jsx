import React, {Component} from 'react';
import PropTypes from 'prop-types';
import TextField from '@material-ui/core/TextField';
import Grid from '@material-ui/core/Grid';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogTitle from '@material-ui/core/DialogTitle';

const initialState = {
    quantity: null,
    description: null,
    value: null
};

class FormServices extends Component {

    state = initialState;

    formHandleChange(name, value)
    {
        const state = {...this.state};
        state[name] = value;
        this.setState(state);
    }

    handleClose = () => {
        this.props.onClose();
    };

    handleSave = () => {
        this.props.onSave(this.state);

    };

    render() {
        return (
            <Dialog
                open={this.props.open}
                keepMounted
            >
                <DialogTitle id="alert-dialog-slide-title">
                    Adicionar/Editar Serviço
                </DialogTitle>
                <DialogContent>
                    <Grid container spacing={8}>
                        <Grid item xs={12} sm={6}>
                            <TextField
                                fullWidth
                                label={'Valor'}
                                value={this.state.value || ''}
                                onChange={e => this.formHandleChange('value', e.target.value)}
                                margin="normal" variant="outlined"
                            />
                        </Grid>

                        <Grid item xs={12} sm={6}>
                            <TextField
                                fullWidth
                                label={'Quantidade'}
                                value={this.state.quantity || ''}
                                onChange={e => this.formHandleChange('quantity', e.target.value)}
                                margin="normal" variant="outlined"
                            />
                        </Grid>

                        <Grid item xs={12} sm={12}>
                            <TextField
                                fullWidth
                                label={'Descrição'}
                                value={this.state.description || ''}
                                onChange={e => this.formHandleChange('description', e.target.value)}
                                margin="normal" variant="outlined"
                                multiline
                                rows={3}
                            />
                        </Grid>
                    </Grid>
                </DialogContent>
                <DialogActions>
                    <Button variant="contained" color="primary" onClick={this.handleSave} className={'btn-save'}>
                        Salvar
                    </Button>

                    <Button variant="contained" onClick={this.handleClose} className={'btn-close'}>
                        Cancelar
                    </Button>
                </DialogActions>
            </Dialog>

        );
    }
}

FormServices.defaultProps = {
    onSave: () => null,
    onClose: () => null,
    open: false
};

FormServices.propTypes = {
    onSave: PropTypes.func,
    onClose: PropTypes.func,
    open: PropTypes.bool
};

export default FormServices;