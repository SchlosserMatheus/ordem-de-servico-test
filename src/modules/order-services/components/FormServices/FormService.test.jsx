import React from 'react';
import { shallow, mount } from 'enzyme';
import FormServices from "./FormServices";
import sinon from 'sinon';

describe('Test <FormService />', () => {

    // it('should render correctly', () => {
    //     const wrapper = shallow(<FormServices/>);
    // });

    //todo test component props

    it('should have component initial state', () => {
        const wrapper = shallow(<FormServices/>);
        const component = wrapper.instance();
        const initialState = {
            quantity: null,
            description: null,
            value: null
        };

        expect(component.state).toEqual(initialState);
    });


    it('should change component state by index', () => {
        const wrapper = shallow(<FormServices/>);
        const component = wrapper.instance();

        const state = {
            quantity: 'quantity',
            description: 'description',
            value: 'value'
        };

        component.formHandleChange('quantity', state.quantity);
        component.formHandleChange('description', state.description);
        component.formHandleChange('value', state.value);

        expect(component.state).toEqual(state);
    });



    it('should emit onSave event when the save button is clicked', () => {
        const onSave = sinon.spy();
        const wrapper = mount(<FormServices onSave={onSave} />);
        const button = wrapper.find('button.btn-save');
        expect(button.length).toEqual(1);

        button.simulate('click');
        expect(onSave.callCount).toEqual(1);
        wrapper.unmount();
    });

    it('should emit onSave event when use the method handleSave', () => {
        const onSave = sinon.spy();
        const wrapper = shallow(<FormServices onSave={onSave} />);
        const component = wrapper.instance();
        component.handleSave();
        expect(onSave.callCount).toEqual(1);
    });

    it('should emit onSave event with state in callback args', () => {
        const onSave = sinon.spy();
        const wrapper = shallow(<FormServices onSave={onSave} />);
        const component = wrapper.instance();
        const args = { description: null, quantity: null, value: null };

        component.handleSave();
        expect(onSave.firstCall.args[0]).toEqual(args);
    });

    //todo verify how to do the test whe pass state for edit mode
    it('should reset state when onSave event is emitted', () => {
        // const onSave = sinon.spy();
        // const wrapper = shallow(<FormServices onSave={onSave} />);
        // const component = wrapper.instance();
        // const args = { description: null, quantity: null, value: null };
        //
        // component.handleSave();
        // expect(onSave.firstCall.args[0]).toEqual(args);
    });


    it('should emit onClose event when use the handleClose method', () => {
        const onClose = sinon.spy();
        const wrapper = shallow(<FormServices onClose={onClose} />);
        const component = wrapper.instance();
        component.handleClose();
        expect(onClose.callCount).toEqual(1);
    });

    it('should emit onClose event when the close button is clicked', () => {

        const onClose = sinon.spy();
        const wrapper = mount(<FormServices onClose={onClose} />);
        const button = wrapper.find('button.btn-close');
        expect(button.length).toEqual(1);

        button.simulate('click');
        expect(onClose.callCount).toEqual(1);
        wrapper.unmount();
    });

    it('check snapshot version', () => {
        const wrapper = shallow(<FormServices/>);
        expect(wrapper).toMatchSnapshot();
    });

    //todo test prop serviceData
    //todo test snapshot component with prop open false

});