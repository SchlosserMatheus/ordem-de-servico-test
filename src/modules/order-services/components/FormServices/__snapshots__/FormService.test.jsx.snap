// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <FormService /> check snapshot version 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FormServices
    onClose={[Function]}
    onSave={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(DialogTitle)
          id="alert-dialog-slide-title"
        >
          Adicionar/Editar Serviço
        </WithStyles(DialogTitle)>,
        <WithStyles(DialogContent)>
          <WithStyles(Grid)
            container={true}
            spacing={8}
          >
            <WithStyles(Grid)
              item={true}
              sm={6}
              xs={12}
            >
              <TextField
                fullWidth={true}
                label="Valor"
                margin="normal"
                onChange={[Function]}
                required={false}
                select={false}
                value=""
                variant="outlined"
              />
            </WithStyles(Grid)>
            <WithStyles(Grid)
              item={true}
              sm={6}
              xs={12}
            >
              <TextField
                fullWidth={true}
                label="Quantidade"
                margin="normal"
                onChange={[Function]}
                required={false}
                select={false}
                value=""
                variant="outlined"
              />
            </WithStyles(Grid)>
            <WithStyles(Grid)
              item={true}
              sm={12}
              xs={12}
            >
              <TextField
                fullWidth={true}
                label="Descrição"
                margin="normal"
                multiline={true}
                onChange={[Function]}
                required={false}
                rows={3}
                select={false}
                value=""
                variant="outlined"
              />
            </WithStyles(Grid)>
          </WithStyles(Grid)>
        </WithStyles(DialogContent)>,
        <WithStyles(DialogActions)>
          <WithStyles(Button)
            className="btn-save"
            color="primary"
            onClick={[Function]}
            variant="contained"
          >
            Salvar
          </WithStyles(Button)>
          <WithStyles(Button)
            className="btn-close"
            onClick={[Function]}
            variant="contained"
          >
            Cancelar
          </WithStyles(Button)>
        </WithStyles(DialogActions)>,
      ],
      "keepMounted": true,
      "onClose": [Function],
      "open": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Adicionar/Editar Serviço",
          "id": "alert-dialog-slide-title",
        },
        "ref": null,
        "rendered": "Adicionar/Editar Serviço",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Grid)
            container={true}
            spacing={8}
          >
            <WithStyles(Grid)
              item={true}
              sm={6}
              xs={12}
            >
              <TextField
                fullWidth={true}
                label="Valor"
                margin="normal"
                onChange={[Function]}
                required={false}
                select={false}
                value=""
                variant="outlined"
              />
            </WithStyles(Grid)>
            <WithStyles(Grid)
              item={true}
              sm={6}
              xs={12}
            >
              <TextField
                fullWidth={true}
                label="Quantidade"
                margin="normal"
                onChange={[Function]}
                required={false}
                select={false}
                value=""
                variant="outlined"
              />
            </WithStyles(Grid)>
            <WithStyles(Grid)
              item={true}
              sm={12}
              xs={12}
            >
              <TextField
                fullWidth={true}
                label="Descrição"
                margin="normal"
                multiline={true}
                onChange={[Function]}
                required={false}
                rows={3}
                select={false}
                value=""
                variant="outlined"
              />
            </WithStyles(Grid)>
          </WithStyles(Grid)>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Grid)
                item={true}
                sm={6}
                xs={12}
              >
                <TextField
                  fullWidth={true}
                  label="Valor"
                  margin="normal"
                  onChange={[Function]}
                  required={false}
                  select={false}
                  value=""
                  variant="outlined"
                />
              </WithStyles(Grid)>,
              <WithStyles(Grid)
                item={true}
                sm={6}
                xs={12}
              >
                <TextField
                  fullWidth={true}
                  label="Quantidade"
                  margin="normal"
                  onChange={[Function]}
                  required={false}
                  select={false}
                  value=""
                  variant="outlined"
                />
              </WithStyles(Grid)>,
              <WithStyles(Grid)
                item={true}
                sm={12}
                xs={12}
              >
                <TextField
                  fullWidth={true}
                  label="Descrição"
                  margin="normal"
                  multiline={true}
                  onChange={[Function]}
                  required={false}
                  rows={3}
                  select={false}
                  value=""
                  variant="outlined"
                />
              </WithStyles(Grid)>,
            ],
            "container": true,
            "spacing": 8,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <TextField
                  fullWidth={true}
                  label="Valor"
                  margin="normal"
                  onChange={[Function]}
                  required={false}
                  select={false}
                  value=""
                  variant="outlined"
                />,
                "item": true,
                "sm": 6,
                "xs": 12,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "fullWidth": true,
                  "label": "Valor",
                  "margin": "normal",
                  "onChange": [Function],
                  "required": false,
                  "select": false,
                  "value": "",
                  "variant": "outlined",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <TextField
                  fullWidth={true}
                  label="Quantidade"
                  margin="normal"
                  onChange={[Function]}
                  required={false}
                  select={false}
                  value=""
                  variant="outlined"
                />,
                "item": true,
                "sm": 6,
                "xs": 12,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "fullWidth": true,
                  "label": "Quantidade",
                  "margin": "normal",
                  "onChange": [Function],
                  "required": false,
                  "select": false,
                  "value": "",
                  "variant": "outlined",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <TextField
                  fullWidth={true}
                  label="Descrição"
                  margin="normal"
                  multiline={true}
                  onChange={[Function]}
                  required={false}
                  rows={3}
                  select={false}
                  value=""
                  variant="outlined"
                />,
                "item": true,
                "sm": 12,
                "xs": 12,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "fullWidth": true,
                  "label": "Descrição",
                  "margin": "normal",
                  "multiline": true,
                  "onChange": [Function],
                  "required": false,
                  "rows": 3,
                  "select": false,
                  "value": "",
                  "variant": "outlined",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(Button)
              className="btn-save"
              color="primary"
              onClick={[Function]}
              variant="contained"
            >
              Salvar
            </WithStyles(Button)>,
            <WithStyles(Button)
              className="btn-close"
              onClick={[Function]}
              variant="contained"
            >
              Cancelar
            </WithStyles(Button)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Salvar",
              "className": "btn-save",
              "color": "primary",
              "onClick": [Function],
              "variant": "contained",
            },
            "ref": null,
            "rendered": "Salvar",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Cancelar",
              "className": "btn-close",
              "onClick": [Function],
              "variant": "contained",
            },
            "ref": null,
            "rendered": "Cancelar",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(DialogTitle)
            id="alert-dialog-slide-title"
          >
            Adicionar/Editar Serviço
          </WithStyles(DialogTitle)>,
          <WithStyles(DialogContent)>
            <WithStyles(Grid)
              container={true}
              spacing={8}
            >
              <WithStyles(Grid)
                item={true}
                sm={6}
                xs={12}
              >
                <TextField
                  fullWidth={true}
                  label="Valor"
                  margin="normal"
                  onChange={[Function]}
                  required={false}
                  select={false}
                  value=""
                  variant="outlined"
                />
              </WithStyles(Grid)>
              <WithStyles(Grid)
                item={true}
                sm={6}
                xs={12}
              >
                <TextField
                  fullWidth={true}
                  label="Quantidade"
                  margin="normal"
                  onChange={[Function]}
                  required={false}
                  select={false}
                  value=""
                  variant="outlined"
                />
              </WithStyles(Grid)>
              <WithStyles(Grid)
                item={true}
                sm={12}
                xs={12}
              >
                <TextField
                  fullWidth={true}
                  label="Descrição"
                  margin="normal"
                  multiline={true}
                  onChange={[Function]}
                  required={false}
                  rows={3}
                  select={false}
                  value=""
                  variant="outlined"
                />
              </WithStyles(Grid)>
            </WithStyles(Grid)>
          </WithStyles(DialogContent)>,
          <WithStyles(DialogActions)>
            <WithStyles(Button)
              className="btn-save"
              color="primary"
              onClick={[Function]}
              variant="contained"
            >
              Salvar
            </WithStyles(Button)>
            <WithStyles(Button)
              className="btn-close"
              onClick={[Function]}
              variant="contained"
            >
              Cancelar
            </WithStyles(Button)>
          </WithStyles(DialogActions)>,
        ],
        "keepMounted": true,
        "onClose": [Function],
        "open": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Adicionar/Editar Serviço",
            "id": "alert-dialog-slide-title",
          },
          "ref": null,
          "rendered": "Adicionar/Editar Serviço",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Grid)
              container={true}
              spacing={8}
            >
              <WithStyles(Grid)
                item={true}
                sm={6}
                xs={12}
              >
                <TextField
                  fullWidth={true}
                  label="Valor"
                  margin="normal"
                  onChange={[Function]}
                  required={false}
                  select={false}
                  value=""
                  variant="outlined"
                />
              </WithStyles(Grid)>
              <WithStyles(Grid)
                item={true}
                sm={6}
                xs={12}
              >
                <TextField
                  fullWidth={true}
                  label="Quantidade"
                  margin="normal"
                  onChange={[Function]}
                  required={false}
                  select={false}
                  value=""
                  variant="outlined"
                />
              </WithStyles(Grid)>
              <WithStyles(Grid)
                item={true}
                sm={12}
                xs={12}
              >
                <TextField
                  fullWidth={true}
                  label="Descrição"
                  margin="normal"
                  multiline={true}
                  onChange={[Function]}
                  required={false}
                  rows={3}
                  select={false}
                  value=""
                  variant="outlined"
                />
              </WithStyles(Grid)>
            </WithStyles(Grid)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(Grid)
                  item={true}
                  sm={6}
                  xs={12}
                >
                  <TextField
                    fullWidth={true}
                    label="Valor"
                    margin="normal"
                    onChange={[Function]}
                    required={false}
                    select={false}
                    value=""
                    variant="outlined"
                  />
                </WithStyles(Grid)>,
                <WithStyles(Grid)
                  item={true}
                  sm={6}
                  xs={12}
                >
                  <TextField
                    fullWidth={true}
                    label="Quantidade"
                    margin="normal"
                    onChange={[Function]}
                    required={false}
                    select={false}
                    value=""
                    variant="outlined"
                  />
                </WithStyles(Grid)>,
                <WithStyles(Grid)
                  item={true}
                  sm={12}
                  xs={12}
                >
                  <TextField
                    fullWidth={true}
                    label="Descrição"
                    margin="normal"
                    multiline={true}
                    onChange={[Function]}
                    required={false}
                    rows={3}
                    select={false}
                    value=""
                    variant="outlined"
                  />
                </WithStyles(Grid)>,
              ],
              "container": true,
              "spacing": 8,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <TextField
                    fullWidth={true}
                    label="Valor"
                    margin="normal"
                    onChange={[Function]}
                    required={false}
                    select={false}
                    value=""
                    variant="outlined"
                  />,
                  "item": true,
                  "sm": 6,
                  "xs": 12,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "fullWidth": true,
                    "label": "Valor",
                    "margin": "normal",
                    "onChange": [Function],
                    "required": false,
                    "select": false,
                    "value": "",
                    "variant": "outlined",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <TextField
                    fullWidth={true}
                    label="Quantidade"
                    margin="normal"
                    onChange={[Function]}
                    required={false}
                    select={false}
                    value=""
                    variant="outlined"
                  />,
                  "item": true,
                  "sm": 6,
                  "xs": 12,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "fullWidth": true,
                    "label": "Quantidade",
                    "margin": "normal",
                    "onChange": [Function],
                    "required": false,
                    "select": false,
                    "value": "",
                    "variant": "outlined",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <TextField
                    fullWidth={true}
                    label="Descrição"
                    margin="normal"
                    multiline={true}
                    onChange={[Function]}
                    required={false}
                    rows={3}
                    select={false}
                    value=""
                    variant="outlined"
                  />,
                  "item": true,
                  "sm": 12,
                  "xs": 12,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "fullWidth": true,
                    "label": "Descrição",
                    "margin": "normal",
                    "multiline": true,
                    "onChange": [Function],
                    "required": false,
                    "rows": 3,
                    "select": false,
                    "value": "",
                    "variant": "outlined",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Button)
                className="btn-save"
                color="primary"
                onClick={[Function]}
                variant="contained"
              >
                Salvar
              </WithStyles(Button)>,
              <WithStyles(Button)
                className="btn-close"
                onClick={[Function]}
                variant="contained"
              >
                Cancelar
              </WithStyles(Button)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Salvar",
                "className": "btn-save",
                "color": "primary",
                "onClick": [Function],
                "variant": "contained",
              },
              "ref": null,
              "rendered": "Salvar",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Cancelar",
                "className": "btn-close",
                "onClick": [Function],
                "variant": "contained",
              },
              "ref": null,
              "rendered": "Cancelar",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
